public/omni/cluster-management/importing-talos-clusters.mdx

```diff
@@ -0,0 +1,202 @@
+Cluster import command tries to detect if a Talos machine was configured using a boot asset from an Image Factory, hence connected to a Schematic. If so, it will try to ensure that Schematic with Omni's connected Image Factory.
+
+If a Talos machine is not configured using an Image Factory, then the validation step will fail by reporting that the Schematic can't be found.
+However, this doesn't necessarily mean that the cluster can't be imported to Omni.
+It's still possible to continue with cluster import in such a scenario by following one option:
+ * Create a Schematic in Omni's Image Factory that matches the Talos machine configuration and use the provided image to upgrade the Talos machine.
+ * Re-run the import command with `--force` flag to skip the validations.
+
+Note that if `--force` option was used, Omni will not be able to ensure that it can successfully offer upgrades for these Talos machines while keeping their existing configuration.
+Omni will still be able to offer upgrades for these machines, but it will create and use a new Schematic to do so.
+If a custom Linux kernel or custom extensions were used while configuring the machine, they will be lost upon upgrade.
+
+This will not affect machines configured by using a boot asset from [Talos releases](https://github.com/siderolabs/talos/releases).
+
+### Config Patches
+
+While importing a cluster to Omni, the idea is to leave the current state of the cluster as is and not to change any configuration on the nodes, unless it's absolutely necessary.
+Cluster Machine config patches are used to achieve this idea.
+
+The import command uses initial talos and kubernetes versions to build the default machine config for each node in the cluster.
+These configs are then compared to the current machine config on each node to build a config patch.
+Each config patch includes the changes made to the machine config since the Talos cluster was created.
+Certain machine config fields are not allowed to be set on Omni managed clusters. These fields are not included in the final config patch.
+
+After a cluster is imported to Omni, the final machine config for each node is built by applying the config patch to the default machine config built using the initial talos and kubernetes versions used on cluster import.
+Redacted version of this machine config can be found under the `Config` tab of the Cluster Machine overview page in the Omni UI, while config patches can be found under `Patches` tab of the same overview page.
+
+### Unlocking the Cluster
+
+Since Omni uses its own Kubernetes apiserver endpoint, when the cluster is unlocked, Omni will update `kube-apiserver`, `kube-scheduler` and `kube-controller-manager` static pod definitions to point to the Omni-managed kubernetes apiserver endpoint.
+
+### Aborting an ongoing Import
```

Comment:
I think the users will find it useful if you describe the potential risks involved with importing a cluster. A question that pops into my mind is "can I just try importing and abort if it goes wrong, or there might be consequences?"

Reply:
Since the backup includes all the machine configs, assumption was that anything Omni does is reversible just by applying backed up machine configs.